includes/wc-product-functions.php

```diff
@@ -1310,3 +1266,27 @@ function wc_deferred_product_sync( $product_id ) {
 
 	$wc_deferred_product_sync[] = $product_id;
 }
+
+/**
+ * Add sorting index data for products.
+ */
+function wc_update_product_lookup_tables() {
+	global $wpdb;
+
+	$result = $wpdb->query(
+		"
+		INSERT IGNORE INTO {$wpdb->wc_product_sorting} (`product_id`, `price`, `min_price`, `max_price`, `average_rating`, `total_sales`)
```

Comment:
I think doing everything in one query might cause issues with the update routine timing out and not completing the transfer of all the data. Best would probably be to do the update in batches at a time?

Reply:
Does SQL timeout if this is a single query?

Asking because in the indexes plugin we also use a single big query.

Reply:
In this case, I'm not too worried about doing everything in a single query as the number of products is relatively small for most stores and we will be writing to a table that is new and thus it will be used only after it is populated. What if we run some tests to see how long does it that to run this single query and then we can decide if it is worth considering other approaches?

@kloon, let me know if I'm missing something. I didn't have time yet to play with this query.

> Asking because in the indexes plugin we also use a single big query.

AFAIK, in the indexes plugin, we don't have an option as there is no other way to add an index.

Reply:
Sounds good to me, I was worried because we broke things up in the post_author PR, however, I now see why we did it there, we were updating an existing table and not a new one. Should be good with one query on a new table.

Reply:
Definitely worth testing on a large database anyway just to be sure.